client/post-editor/editor-featured-image/index.jsx

```diff
@@ -125,25 +116,17 @@ const EditorFeaturedImage = React.createClass( {
 			'is-assigned': !! PostUtils.getFeaturedImageId( this.props.post )
 		} );
 
-		if ( this.props.editable ) {
-			return (
-				<AccordionSection className={ classes }>
-					{ this.renderMediaModal() }
-					<EditorDrawerWell
-						icon="image"
-						label={ this.translate( 'Set Featured Image' ) }
-						onClick={ () => this.toggleMediaModal( 'show' ) }
-						onRemove={ this.removeImage }>
-						{ this.renderCurrentImage() }
-					</EditorDrawerWell>
-				</AccordionSection>
-			);
-		}
-
 		return (
-			<div className={ classes }>
-				{ this.renderCurrentImage() }
-			</div>
+			<Button
+				onClick={ () => this.toggleMediaModal( 'show' ) }
```

Comment:
>Create a `showMediaModal` function to avoid the fat arrow here?

Indeed, this works much better. Removed `toggleMediaModal` in favor of two separate `showMediaModal` and `hideMediaModal`.